django/db/models/query.py

```diff
@@ -1711,7 +1715,68 @@ def extra(
         if self.query.is_sliced:
             raise TypeError("Cannot change a query once a slice has been taken.")
         clone = self._chain()
-        clone.query.add_extra(select, select_params, where, params, tables, order_by)
+        if tables:
+            clone.query.add_extra_tables(tables)
+        if select and (not clone.query.values_select or clone.query.values_select_all):
+            warnings.warn(
+                "extra(select) usage is deprecated, use annotate() with RawSQL "
+                "instead.",
+                category=RemovedInDjango60Warning,
+                stacklevel=3,
+            )
+            # XXX: This is mising the logic to always place extras at the begining
+            # of the select clause.
```

Comment:
This is assuming we want to preserve this behavior during the deprecation period as [it was never documented](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#extra) and only becomes a problem when using `extra().values_list()` which I would argue is bad practice.